### PR TITLE
fix(isArrayLike): reject functions because functions have "length"

### PR DESCRIPTION
**Pull request by johnlindquist**
**Description:**
Added a `typeof x !== "function"` check to `isArrayLike`. `isArrayLike` was only checking for `length` and functions have a `length` property. This prevented functions from properly falling through to this case:

https://github.com/ReactiveX/rxjs/blob/6fd1fd7719e690d93ba3205db8b6b6e9e0990579/src/internal/util/subscribeTo.ts#L31

### CODE REVIEW

File: spec/observables/from-spec.ts [hunk 1]
```diff
@@ -16,13 +17,13 @@ function getArguments<T>(...args: T[]) {
...
-      .concatMap((x, i) => of(x).delay(i === 0 ? 0 : 20, rxTestScheduler));
-    const expected = 'x-y-(z|)';
-    expectObservable(e1).toBe(expected, {x: 10, y: 20, z: 30});
-  });
+    ('should create an observable from an array', () => {
+      const e1 = from([10, 20, 30])
+        // for the purpose of making a nice diagram, spread out the synchronous emissions
+        .concatMap((x, i) => of(x).delay(i === 0 ? 0 : 20, rxTestScheduler));
+      const expected = 'x-y-(z|)';
+      expectObservable(e1).toBe(expected, { x: 10, y: 20, z: 30 });
```
Comment (benlesh) [inline spec/observables/from-spec.ts]: Can we undo the formatting changes?

File: spec/observables/from-spec.ts [hunk 2]
```diff
@@ -118,5 +119,26 @@ describe('from', () => {
         );
       expect(nextInvoked).to.equal(false);
     });
+    it(`should accept a function`, (done) => {
```
Comment (benlesh) [inline spec/observables/from-spec.ts]: `from` should not accept a function.

`from((...args) => subject.next(...args))` should not work and should error.

The change to `isArrayLike` I can understand, I'm uncertain about this test.

Comment (johnlindquist) [reply spec/observables/from-spec.ts]: if `from` accepts a `Promise`, shouldn't it be able to accept a `handler`?

File: spec/observables/from-spec.ts (hunk 1 above)
Comment (johnlindquist) [reply spec/observables/from-spec.ts]: done

File: spec/observables/from-spec.ts (hunk 2 above)
Comment (benlesh) [reply spec/observables/from-spec.ts]: I see, the function has `Symbol.observable` on it. I missed that. Apologies.

**Comment by benlesh** [comment]
Thanks @johnlindquist <3

**Pull request merged**